native_client/ctcdecode/scorer.cpp

```diff
@@ -262,18 +268,20 @@ void Scorer::fill_dictionary(const std::vector<std::string>& vocabulary, bool ad
    * can greatly increase the size of the FST
    */
   fst::RmEpsilon(&dictionary);
-  fst::StdVectorFst* new_dict = new fst::StdVectorFst;
+  std::unique_ptr<fst::StdVectorFst> new_dict(new fst::StdVectorFst);
 
   /* This makes the FST deterministic, meaning for any string input there's
    * only one possible state the FST could be in.  It is assumed our
    * dictionary is deterministic when using it.
    * (lest we'd have to check for multiple transitions at each state)
    */
-  fst::Determinize(dictionary, new_dict);
+  fst::Determinize(dictionary, new_dict.get());
 
   /* Finds the simplest equivalent fst. This is unnecessary but decreases
    * memory usage of the dictionary
    */
-  fst::Minimize(new_dict);
-  this->dictionary.reset(new_dict);
+  fst::Minimize(new_dict.get());
+
+  std::unique_ptr<FstType> converted(new FstType(*new_dict));
```

Comment:
note to self: document fst impl conversion happening here

Reply:
yes, please